src/H5Fint.c

```diff
@@ -1603,8 +1606,12 @@ H5F__dest(H5F_t *f, hbool_t flush)
     /* Free the non-shared part of the file */
     f->open_name   = (char *)H5MM_xfree(f->open_name);
     f->actual_name = (char *)H5MM_xfree(f->actual_name);
-    if (f->vol_obj && H5VL_free_object(f->vol_obj) < 0)
-        HDONE_ERROR(H5E_FILE, H5E_CANTDEC, FAIL, "unable to free VOL object")
+    if (f->vol_obj) {
```

Comment:
Agree - this is necessary.   (Although moving the "f->vol_obj = NULL;" into the new 'if' block would be slightly cleaner.

Reply:
(You may also want to leave a comment in H5F__close_cb() that unwrapping the object before calling H5VL_free_object is unnecessary since it wasn't wrapped)

Reply:
Also done. Hopefully this is the only place where this weird one-off unwrapping is needed. It seems like everywhere else that H5VL_free_object is used does the correct thing, but there might be more unwrapping issues to be discovered when passthrough connectors do complicated things.